src/Tools/InternalHelpers.php

```diff
@@ -0,0 +1,76 @@
+<?php
+declare(strict_types=1);
+
+namespace PHPSu\Tools;
+
+use PHPSu\Controller;
+use PHPSu\Process\CommandExecutor;
+use PHPSu\Process\Process;
+
+final class InternalHelpers
+{
+
+    private const FALLBACK_VERSION = 'console';
+
+    public function getCurrentPHPSUVersion(): string
+    {
+        if ($this->isVendorPackage()) {
+            return $this->getPhpSuVersionFromVendor();
+        }
+        if (EnvironmentUtility::isGitInstalled()) {
+            return trim($this->getPhpSuVersionFromGitCommand());
+        }
+        if ($this->isGitFolderAvailable()) {
+            return $this->getPhpSuVersionFromGitFolder();
+        }
+        return self::FALLBACK_VERSION;
+    }
+
+    private function isVendorPackage(): bool
+    {
+        return false;
+    }
+
+    private function getPhpSuVersionFromVendor(): string
+    {
+        return '';
+    }
+
+    private function getPhpSuVersionFromGitFolder(): string
```

Comment:
we need only one of ``getPhpSuVersionFromGitFolder`` or ``getPhpSuVersionFromGitCommand``

Reply:
right, one of them i am going to drop but i did not test them 100% to be sure which one is working to the exact need